src/lib/apexPmd.ts

```diff
@@ -240,13 +227,17 @@ export class ApexPmd {
   createDiagnostic(result: PmdResult): vscode.Diagnostic {
     const { priorityErrorThreshold, priorityWarnThreshold } = this.config;
     const lineNum = parseInt(result.line) - 1;
-
-    const uri = `https://pmd.github.io/latest/pmd_rules_apex_${result.ruleSet
-      .split(' ')
-      .join('')
-      .toLowerCase()}.html#${result.rule.toLowerCase()}`;
+    let uri = '';
     const msg = `${result.description} (rule: ${result.ruleSet}-${result.rule})`;
-
+    const externalURL = result.externalURL;
+    if (externalURL != undefined) {
+      uri = externalURL.startsWith('http') ? externalURL : 'https://' + externalURL;
+    } else {
+      uri = `https://pmd.github.io/latest/pmd_rules_apex_${result.ruleSet
+        .split(' ')
+        .join('')
+        .toLowerCase()}.html#${result.rule.toLowerCase()}`;
```

Comment:
I think we should remove these lines - if there is no externalURL available in PMD's report, then it is definitively not a default rule from PMD and the URL that is constructed here will always be wrong (as before for custom rules, see #162).

Is there some "safe" value we can return instead, s.th. like "no info available"?